TheMealDB/TheMealDB/Moduls/CategroryScreen/Controller/CategoryController.swift

```diff
@@ -40,22 +48,89 @@ class CategoryController: UIViewController {
                     return
                 }
                 self.categories = category
+                self.filterCategory = category
+            case .failure(let error): print(error as Any)
+            }
+        }
+        let mealRepository: MealRepository = MealRepositoryImp(api: APIService.share)
+        mealRepository.getMeal { [weak self] result in
+            guard let `self` = self else { return }
+            switch result {
+            case .success(let meal):
+                guard let meal = meal?.meals else {
+                    return
+                }
+                self.meals = meal
             case .failure(let error): print(error as Any)
             }
         }
     }
+    
+    private func setUpNavigationBar() {
+        self.navigationController?.navigationBar.isTranslucent = true
+        self.navigationController?.navigationBar.setBackgroundImage(UIImage(), for: .default)
+        self.navigationController?.navigationBar.shadowImage = UIImage()
+        self.navigationItem.title = Resource.Navigation.categoriesTitle
+        let searchBarButton = UIBarButtonItem(image: Resource.Images.searchButton,
+                                              style: .plain,
+                                              target: self, action: #selector(setUpSearchBar))
+        self.navigationItem.rightBarButtonItem = searchBarButton
+    }
+    
+    @objc func setUpSearchBar() {
+        let searchBar = UISearchBar()
+        searchBar.delegate = self
+        searchBar.showsCancelButton = true
+        searchBar.placeholder = "Search Categrory"
+        self.navigationItem.titleView = searchBar
+    }
 }
 
-extension CategoryController: UITableViewDataSource, UITableViewDelegate {
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return categories.count
+extension CategoryController: UISearchBarDelegate {
+    //  MARK: - Setup Action
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        guard searchText.isEmpty == false else {
+            filterCategory = categories
+            return
+        }
+        filterCategory = categories.filter({ category -> Bool in
```

Comment:
sử dụng trailing closure để tối giản cú pháp ở đây
```suggestion
         filterCategory = categories.filter { category -> Bool in
```